drivers/flash/spi_nor.c

```diff
@@ -540,6 +540,45 @@ static int spi_nor_wrsr(const struct device *dev,
 	return ret;
 }
 
+#define MX25R_HIGH_PERF_MODE BIT(1)
+
+#ifdef CONFIG_SPI_NOR_HIGH_PERF_MODE
+static int enable_high_perf_mode(const struct device *dev, uint8_t jedec_id[])
+{
+	int sr, ret;
+	uint8_t data[3];
+
+	/* Only supported on Macronix MX25R Ultra Low Power series. */
+	if (jedec_id[0] != 0xc2 || jedec_id[1] != 0x28) {
+		return -ENOTSUP;
+	}
+
+	acquire_device(dev);
+
+	/* The configuration register bytes on the Macronix MX25R devices are set using
+	 * the Write Status Register command. First read out the current status register.
+	 */
+	sr = spi_nor_rdsr(dev);
+	if (sr < 0) {
+		return sr;
+	}
+
+	data[0] = sr;
+	data[1] = 0x0;			/* Configuration register 1 */
+	data[2] = MX25R_HIGH_PERF_MODE;	/* Configuration register 2 */
+
+	ret = spi_nor_cmd_write(dev, SPI_NOR_CMD_WREN);
+	if (ret == 0) {
+		ret = spi_nor_access(dev, SPI_NOR_CMD_WRSR, NOR_ACCESS_WRITE, 0, data,
+			sizeof(data));
+		spi_nor_wait_until_ready(dev);
+	}
+	release_device(dev);
+
+	return ret;
+}
+#endif /* SPI_NOR_HIGH_PERF_MODE */
```

Comment:
`#endif /* CONFIG_SPI_NOR_HIGH_PERF_MODE */` will make it match the other endif's. See also L1153.

Reply:
Sorry, I don't understand the problem it causes nor what I should change. Isn't the comment just an indication to the reader what #ifdef the #endif belongs to? Is an editor interpreting that incorrectly?

Reply:
Noticed now that I was missing `CONFIG_` in the comment. Could that be related?

Reply:
You're right, it's purely for the user and it doesn't matter for the program (and not my editor either). It just seems like all the other `#endif`s comments start with `CONFIG_` and I'm not sure how strict code owners are on this kind of thing.
```suggestion
#endif /* CONFIG_SPI_NOR_HIGH_PERF_MODE */
```

Reply:
Yes, it was a mistake on my end. I first (before committing) forgot to use `CONFIG_` in the `#ifdef` as well, but forgot to update the comment when fixing the `#ifdef`. Have updated the comments now.